test/integration/puppetlabs/services/jruby/puppet_environments_int_test.clj

```diff
@@ -193,3 +193,41 @@
         (let [catalogs (get-catalog-from-each-jruby borrow-jruby-fn return-jruby-fn)]
           (is (= 0 (num-catalogs-containing catalogs "Notify" "hello1")))
           (is (= 2 (num-catalogs-containing catalogs "Notify" "hello2"))))))))
+
+(deftest ^:integration single-environment-flush-integration-test
+  (testing "a single environment is flushed after marking expired"
+    (write-site-pp-file "include foo")
+    (write-foo-pp-file "class foo { notify {'hello1': } }")
+    (bootstrap/with-puppetserver-running app {:jruby-puppet
+                                              {:max-active-instances 1}}
+      ;;; Validate that the catalog has `hello1`
+      (let [catalog1 (get-catalog)]
+        (is (catalog-contains? catalog1 "Notify" "hello1"))
+        (is (not (catalog-contains? catalog1 "Notify" "hello2"))))
+
+      ;; Now we modify the class definition to have a 'hello2' notify,
+      ;; instead of 'hello1'.
+      (write-foo-pp-file "class foo { notify {'hello2': } }")
+
```

Comment:
When I wrote my comment above, I hadn't yet realized that the first request below specified a different environment - given that, I don't feel so strongly about adding the extra validation I mentioned - I'd be fine adding it or not.